src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -41,18 +42,26 @@
   @Override
   /* Show courses. */
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Course> courses = getHelper(request);
+    JSONObject results = getHelper(request);
 
-    String coursesJson = new Gson().toJson(courses);
+    // String resultsJson = new Gson().toJson(results);
     response.setContentType("application/json;");
-    response.getWriter().println(coursesJson);
+    response.getWriter().println(results);
   }
 
   /* Create list of courses given request. Used for testing. */
-  public List<Course> getHelper(HttpServletRequest request) {
+  public JSONObject getHelper(HttpServletRequest request) {
+    String success = "good";
     List<Filter> filters = getFilters(request);
-    List<Entity> results = getResults(filters);
-    List<Course> courses = getCourses(results);
+    List<Entity> results = getResults(filters, false);
+    if (results.size() == 0) {
+      results = getResults(filters, true);
```

Comment:
Would you not have to check that the results is not empty before you can say success or am I misunderstanding the code?
`if(results.isEmpty()){`
  `  results = getResults(filters,true);`
  `  if(!results.isEmpty()){`
     `    success = "okay"`
 `   }`
`}`

Reply:
Success represents how successful the search was (exact match  = good, similar matches = okay, no matches = bad). I use it to print a message when displaying results, if applicable.